// app/controllers/item.controller.js
const Item = require("../models/item.model.js");

exports.create = (req, res) => {
  if (!req.body) {
    res.status(400).send({
      message: "Content can not be empty!"
    });
  }

  const item = new Item({
    name: req.body.name,
    qty: req.body.qty,
    amount: req.body.amount
  });

  Item.create(item, (err, data) => {
    if (err)
      res.status(500).send({
        message:
          err.message || "Error."
      });
    else res.send(data);
  });
};


exports.findAll = (req, res) => {
  Item.getAll((err, data) => {
    if (err)
      res.status(500).send({
        message:
          err.message || "error."
      });
    else res.send(data);
  });
};

exports.findOne = (req, res) => {
  Item.findById(req.params.itemId, (err, data) => {
    if (err) {
      if (err.kind === "not_found") {
        res.status(404).send({
          message: `Missing ${req.params.itemId}.`
        });
      } else {
        res.status(500).send({
          message: "Error on " + req.params.itemId
        });
      }
    } else res.send(data);
  });
};

exports.update = (req, res) => {
  // Validate Request
  if (!req.body) {
    res.status(400).send({
      message: "Fill blanks"
    });
  }

  Item.updateById(
    req.params.itemId,
    new Item(req.body),
    (err, data) => {
      if (err) {
        if (err.kind === "not_found") {
          res.status(404).send({
            message: `Missing ${req.params.itemId}.`
          });
        } else {
          res.status(500).send({
            message: "Error on " + req.params.itemId
          });
        }
      } else res.send(data);
    }
  );
};

exports.delete = (req, res) => {
  Item.remove(req.params.itemId, (err, data) => {
    if (err) {
      if (err.kind === "not_found") {
        res.status(404).send({
          message: `Missing ${req.params.itemId}.`
        });
      } else {
        res.status(500).send({
          message: "Error " + req.params.itemId
        });
      }
    } else res.send({ message: `item was deleted successfully!` });
  });
};

exports.deleteAll = (req, res) => {
  Item.removeAll((err, data) => {
    if (err)
      res.status(500).send({
        message:
          err.message || "Error"
      });
    else res.send({ message: `All deleted!` });
  });
};
